department-of-veterans-affairs/va.gov-cms
# VACMS-17116: Create Views result condition plugin

## Description

Relates to #17116

Background:
Because we are leveraging the ECA (Event, Condition, Action) module for the notifications framework, we need a way to query a View for results before acting on them. So this PR creates:

- A new ECA Condition plugin: Views Result. This plugin queries a configured View and returns TRUE only if there are results.
- A new Views Display plugin: ECA Result. This plugin doesn't provide any functionality on its own but the Views Result Condition Plugin will allow selecting only Views which have this ECA Result Display type. This works similarly to how Entity Reference fields can be configured to allow entities only from a View with an Entity Reference Source Views display to be referenced.


## Testing done
Tested using various node types and Views.

## Screenshots
![Screenshot 2024-02-27 at 1 26 08 PM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/221539/dace2000-c41e-4437-a87c-90eb14378f41)
![Screenshot 2024-02-27 at 1 25 50 PM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/221539/0412d8ff-bd07-48e7-8ad7-ffd6fed6e9c4)
![Screenshot 2024-02-27 at 1 27 43 PM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/221539/5cefa705-06ea-40ba-99b7-c41c77d80cc9)

## QA steps

As an administrator:
1. Visit https://pr17357-6w0agqkcuvtjtypasyuin5rrkkjvr8sj.ci.cms.va.gov/admin/config/workflow/eca/expiring_fwb/edit
2. Under Workflow, click the 'Enabled' checkbox, and click the Save button.
3. In the Admin Toolbar, to the left of the 'Content' menu, under the VA menu, select 'Run cron'
4. Wait for the page to reload
    - [x] Verify there is a message at the top of the page that reads 'The Scheduled Notifications: Full Width Alert View has results.'
5. Visit https://pr17357-6w0agqkcuvtjtypasyuin5rrkkjvr8sj.ci.cms.va.gov/admin/structure/views/view/scheduled_notifications_full_width_alert/edit/eca_result_1
6. In the Pager section, click the '10 items' link.
7. In the modal, set the Offset to 11 and click Apply
8. Save the View
9. In the Admin Toolbar, to the left of the 'Content' menu, under the VA menu, select 'Run cron'
10. Wait for the page to reload
    - [x] Verify there is NO message at the top of the page that reads 'The Scheduled Notifications: Full Width Alert View 

### Definition of Done


### Select Team for PR review
